auth_api_key/models/auth_api_key.py

```diff
@@ -0,0 +1,62 @@
+# Copyright 2018 ACSONE SA/NV
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import _, api, fields, models, tools
+from odoo.exceptions import AccessError, ValidationError
+from odoo.tools import consteq
+
+
+class AuthApiKey(models.Model):
+    _name = "auth.api.key"
+    _description = "API Key"
+
+    name = fields.Char(required=True)
+    key = fields.Char(
+        required=True,
+        help="""The API key. Enter a dummy value in this field if it is
+        obtained from the server environment configuration.""",
+    )
+    user_id = fields.Many2one(
+        comodel_name="res.users",
+        string="User",
+        required=True,
+        help="""The user used to process the requests authenticated by
+        the api key""",
+    )
+
+    _sql_constraints = [("name_uniq", "unique(name)", "Api Key name must be unique.")]
+
+    @api.model
+    def _retrieve_api_key(self, key):
+        return self.browse(self._retrieve_api_key_id(key))
+
+    @api.model
+    @tools.ormcache("key")
+    def _retrieve_api_key_id(self, key):
+        if not self.env.user.has_group("base.group_system"):
+            raise AccessError(_("User is not allowed"))
+        for api_key in self.search([]):
+            if api_key.key and consteq(key, api_key.key):
+                return api_key.id
+        raise ValidationError(_("The key %s is not allowed") % key)
+
+    @api.model
+    @tools.ormcache("key")
+    def _retrieve_uid_from_api_key(self, key):
+        return self._retrieve_api_key(key).user_id.id
+
+    def _clear_key_cache(self):
+        self.env.registry.clear_cache()
```

Comment:
Unit tests were failing before because clearing the orm cache using `clear_cache()` is not available anymore in 17.0 on the cache object. It was removed in this [commit](https://github.com/odoo/odoo/commit/595aa248433246959a5fa9288e477091701c6a35#diff-3d5ccf8623cd4456c775fef15f0cc973b6f88f05c0ce3b04ab113324e589b851L60) with the introduction of multiple caches. Needed to call `env.registry.clear_cache()` instead. Might be useful for other migrations.

Reply:
@chrisb-c01 Thanks for the details!